internal/monitoring/monitoring.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2016 Canonical Ltd.
+
+package monitoring
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	requestDuration = prometheus.NewSummaryVec(prometheus.SummaryOpts{
+		Namespace: "charmstore",
+		Subsystem: "handler",
+		Name:      "request_duration",
+		Help:      "The duration of a web request.",
+	}, []string{"path_pattern"})
+	uploadProcessingDuration = prometheus.NewSummary(prometheus.SummaryOpts{
+		Namespace: "charmstore",
+		Subsystem: "archive",
+		Name:      "processing_duration",
+		Help:      "The processing duration of a charm upload.",
+	})
```

Comment:
The units should be specified somewhere unless there is an implicit understanding that they are in seconds. Explicit is still better than implicit.